code/go/0chain.net/blobbercore/handler/object_operation_grpc_handler.go

```diff
@@ -0,0 +1,238 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/blobbergrpc"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/convert"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/filestore"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/readmarker"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/reference"
+	"github.com/0chain/blobber/code/go/0chain.net/core/common"
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+	"strconv"
+)
+
+func (b *blobberGRPCService) DownloadFile(ctx context.Context, r *blobbergrpc.DownloadFileRequest) (
+	*blobbergrpc.DownloadFileResponse, error) {
+
+	allocationTx := r.Allocation
+	allocationObj, err := b.storageHandler.verifyAllocation(ctx, allocationTx, false)
```

Comment:
Length of client id check has to be before this 
```

	if len(clientID) == 0 {
		return nil, common.NewError("download_file", "invalid client")
	}
```

Reply:
I did not read this entire file - but can you reduce the changes between this function and original.